test/controllers/api/v2/debs_controller_test.rb

```diff
@@ -0,0 +1,148 @@
+require "katello_test_helper"
+
+module Katello
+  class Api::V2::DebsControllerTest < ActionController::TestCase
+    def models
+      @repo = katello_repositories(:debian_10_amd64)
+      @version = ContentViewVersion.first
+      @deb = katello_debs(:one)
+      @host = hosts(:one)
+      Pulp::Deb.any_instance.stubs(:backend_data).returns({})
+    end
+
+    def setup
+      setup_controller_defaults_api
+      @request.env['HTTP_ACCEPT'] = 'application/json'
+      @request.env['CONTENT_TYPE'] = 'application/json'
+      models
+      setup_product_permissions
+    end
+
+    def test_index
+      get :index, params: { :repository_id => @repo.id }
+
+      assert_response :success
+      assert_template "katello/api/v2/debs/index"
+
+      get :index, params: { :content_view_version_id => @version.id }
+
+      assert_response :success
+      assert_template "katello/api/v2/debs/index"
+    end
+
+    def test_index_with_environment_id
+      environment = KTEnvironment.first
+      KTEnvironment.expects(:readable).returns(stub(:find_by => environment))
+
+      get :index, params: { :environment_id => environment.id }
+
+      assert_response :success
+      assert_template "katello/api/v2/debs/index"
+    end
+
+    def test_index_parameters
+      get :index
+
+      assert_response :success
+    end
+
+    def test_index_with_applicability
+      response = get :index, params: { :host_id => @host.id }
+
+      assert_response :success
+
+      ids = JSON.parse(response.body)['results'].map { |p| p['id'] }
+      assert_includes ids, @deb.id
+    end
+
+    def test_index_with_upgradability
+      response = get :index, params: { :host_id => @host.id, :packages_restrict_upgradable => true }
+
+      assert_response :success
+      ids = JSON.parse(response.body)['results'].map { |p| p['id'] }
+      refute_includes ids, @deb.id
+    end
+
+    def test_index_with_available_for_content_view_version
+      response = get :index, params: { :content_view_version_id => @version.id, :available_for => 'content_view_version' }
+
+      assert_response :success
+      ids = JSON.parse(response.body)['results'].map { |p| p['id'] }
+      assert_includes ids, @deb.id
+    end
+
+    # TODO: not yet supported
+    #def test_index_with_latest
+    #  response = get :index, params: { :packages_restrict_latest => true }
+
+    #  assert_response :success
+    #  ids = JSON.parse(response.body)['results'].map { |p| p['id'] }
+    #  assert_includes ids, katello_debs(:one_new).id
+    #  refute_includes ids, @deb.id
+    #end
```

Comment:
(Reiterating/expanding my answers from the call this morning)

I think we can delete this test

Reply:
I think there is a problem with the fixtures that I do not see right now :frowning_face:

Reply:
Actually it was a bug that is fixed now (see https://github.com/Katello/katello/pull/8277#issuecomment-922876772)